lib/moneta/proxy.rb

```diff
@@ -22,7 +22,7 @@ def each_key(&block)
       raise NotImplementedError, "each_key is not supported on this proxy" \
         unless supports? :each_key
 
-      return enum_for(:each_key) { adapter.each_key.size } unless block_given?
```

Comment:
@asppsa so what do you think of this? Is it worth it? Most of the time the adapters will be used via some proxy (e.g. transformers) so I though it would make sense to honor the "eager size" set in the block, this is for the when someone would use the "size" method, see Object#enum_for:
> If a block is given, it will be used to calculate the size of the enumerator without the need to iterate it (see Enumerator#size).

Without this I think size will be always be nil when a proxy is used.